extensions/ql-vscode/src/cli.ts

```diff
@@ -682,19 +687,56 @@ export class CodeQLCliServer implements Disposable {
 
   async interpretBqrs(metadata: QueryMetadata, resultsPath: string, interpretedResultsPath: string, sourceInfo?: SourceInfo): Promise<sarif.Log> {
     await this.runInterpretCommand(SARIF_FORMAT, metadata, resultsPath, interpretedResultsPath, sourceInfo);
-
-    let output: string;
-    try {
-      output = await fs.readFile(interpretedResultsPath, 'utf8');
-    } catch (e) {
-      const rawMessage = e.stderr || e.message;
-      const errorMessage = rawMessage.startsWith('Cannot create a string')
-        ? `SARIF too large. ${rawMessage}`
-        : rawMessage;
-      throw new Error(`Reading output of interpretation failed: ${errorMessage}`);
-    }
     try {
-      return JSON.parse(output) as sarif.Log;
+      // Parse the SARIF file into token streams, filtering out only the results array.
+      const p = parser();
+      const pipeline = chain([
+        fs.createReadStream(interpretedResultsPath),
+        p,
+        pick({filter: 'runs.0.results'}),
+        verifier()
+      ]);
+
+      // Creates JavaScript objects from the token stream
+      const asm = Assembler.connectTo(pipeline);
+
+      // Returns a constructed Log object with the results or an empty array if no results were found.
+      // If the parser fails for any reason, it will reject the promise.
+      return await new Promise((resolve, reject) => {
+        pipeline.on('error', (error) => {
+          reject(error);
+        });
+
+          asm.on('done', (asm) => {
+            const dummyTool : sarif.Tool = {driver: {name: ''}};
+            if (asm.current) {
+              const log : sarif.Log = {
+                version: '2.1.0', 
+                runs: [
+                  { 
+                    tool: dummyTool, 
+                    results: asm.current
+                  }
+                ]
+              };
+
+              resolve(log);
+            } else {
+              const log : sarif.Log = {
+                version: '2.1.0', 
+                runs: [
+                  { 
+                    tool: dummyTool, 
+                    results: []
+                  }
+                ]
+              };
```

Comment:
Here's a simpler solution (but the formatting is off):

```suggestion
              const log : sarif.Log = {
                version: '2.1.0', 
                runs: [
                  { 
                    tool: dummyTool, 
                    results: asm.current ?? []
                  }
                ]
              };

              resolve(log);
 ``